### ISSUE TITLE
Support for flexible input dimension

### ISSUE THREAD

**Issue by seongwook-ham** [facebookresearch/TensorComprehensions#204]
when using autotuned cached option for tc compiled function, now input size is fixed.
in training neural network, it is common that batchsize is (2^x).
but test set is commonly not divided by batchsize.
so when i want to use tc function practically, we have to autotune function twice for train and test purpose. also when number of gpu or batchsize is changed, or transfer learning is needed, re-autotuning is needed. it is very time consuming.
for these reasons, we need flexible feature dimension.

**Comment by prigoyal** [comment]
Hi @seongwook-ham, thank for the request and detailed explanation. This is also being discussed in the issue https://github.com/facebookresearch/TensorComprehensions/issues/46 does this look same? :)

**Comment by seongwook-ham** [comment]
yes it is same issue

**Comment by prigoyal** [comment]
thanks, we'll close this out as duplicate and the first issue will track the progress :)

**Comment by nicolasvasilache** [comment]
@Roadwalker, please also see my latest reply in #46

**Comment by nicolasvasilache** [comment]
@seongwook-ham see if #225 starts addressing your needs. The code is still JIT'ed and will take a few seconds for various parameter sizes but autotuning results can be easily reused.

Note however that with TC as with any library there is no magic: if your problem sizes varies widely and you want the best performance **you need different** implementations. This is one of the reason the binary size for common libraries is so big: multi-versioning. With TC you can choose to autotune or not (and reuse previously found best options) but you determine the boundaries in the size spaces (by reusing the options you want). Automating this is the focus of work that will start soon but won't pan out for at least 6 months.

If you have real world parametric needs where you see many different versions for 1 parameter (i.e. > 100) then we probably can give you 1-3 parameters in a relatively short term but it would be great to work on a concrete example.
